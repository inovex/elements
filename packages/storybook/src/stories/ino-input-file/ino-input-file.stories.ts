import { useEffect } from '@storybook/client-api';
import { html } from 'lit-html';

export default {
  title: 'Input/ino-input-file',
  component: 'ino-input-file',
  decorators: [
    story => {
      useEffect(() => {
        const eventHandler = function (e) {
          const el = e.target;
          if (el.tagName.toLowerCase() !== 'ino-input-file') {
            return;
          }

          const fileNames = e.detail.files
            .map((f) => [f.name, f.type, f.size + ' bytes'].join(', '))
            .join('\n');
          alert(fileNames);
        };

        document.addEventListener('changeFile', eventHandler);
        return () => document.removeEventListener('changeFile', eventHandler);
      });
      return story();
    }
  ],
  parameters: {
    actions: {
      handles: ['changeFile .customizable-input'],
    },
  },
};

export const Playground = (args) => html`
  <ino-input-file
    class="customizable-input"
    accept="${args.accept}"
    autofocus="${args.autofocus}"
    disabled="${args.disabled}"
    multiple="${args.multiple}"
    required="${args.required}"
    label="${args.label}"
    label-selected="${args.labelSelected}"
    drag-and-drop="${args.dragAndDrop}"
    drag-and-drop-text="${args.dragAndDropText}"
    drag-and-drop-secondary-text="${args.dragAndDropSecondaryText}"
  >
  </ino-input-file>
`;
Playground.args = {
  accept: '',
  autoFocus: false,
  disabled: false,
  multiple: false,
  required: false,
  label: 'Select a file to upload',
  labelSelected: 'selected',
  dragAndDrop: false,
  dragAndDropText: 'Drag your files here',
  dragAndDropSecondaryText: 'or'
};

export const DragAndDrop = () => html`
  <ino-input-file
    multiple
    drag-and-drop
  ></ino-input-file>
`;
