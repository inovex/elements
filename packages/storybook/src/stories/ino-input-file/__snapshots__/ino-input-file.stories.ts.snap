// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input/ino-input-file DragAndDrop smoke-test 1`] = `
<div id="root-inner">
  <ino-input-file class="customizable-input hydrated"
                  accept
                  autofocus="false"
                  disabled="false"
                  multiple="false"
                  required="false"
                  label="Select a file to upload"
                  drag-and-drop="true"
                  drag-and-drop-text="Drag your files here"
                  drag-and-drop-secondary-text="or"
  >
    <div class="ino-input-file__dnd">
      <div class="ino-input-file__dnd-text">
        <label>
          Drag your files here
        </label>
        <label>
          or
        </label>
      </div>
      <ino-button class="ino-input-file__button hydrated"
                  icon-leading
      >
        <button class="button button__variant--filled button__icon--leading"
                name="file-paths"
                type="button"
        >
          <span class="icon__wrapper">
            <ino-icon slot="icon-leading"
                      class="ino-icon--color-scheme-primary hydrated"
            >
            </ino-icon>
          </span>
          <span class="button__label">
            Select a file to upload
          </span>
        </button>
      </ino-button>
      <input class="ino-input-file__native-element"
             accept
             type="file"
             aria-hidden="true"
      >
    </div>
  </ino-input-file>
</div>
`;

exports[`Input/ino-input-file Playground smoke-test 1`] = `
<div id="root-inner">
  <ino-input-file class="customizable-input hydrated"
                  accept
                  autofocus="false"
                  disabled="false"
                  multiple="false"
                  required="false"
                  label="Select a file to upload"
                  drag-and-drop="false"
                  drag-and-drop-text="Drag your files here"
                  drag-and-drop-secondary-text="or"
  >
    <div class="ino-input-file__composer">
      <ino-button class="ino-input-file__button hydrated"
                  icon-leading
      >
        <button class="button button__variant--filled button__icon--leading"
                name="file-paths"
                type="button"
        >
          <span class="icon__wrapper">
            <ino-icon slot="icon-leading"
                      class="hydrated ino-icon--color-scheme-primary"
            >
            </ino-icon>
          </span>
          <span class="button__label">
            Select a file to upload
          </span>
        </button>
      </ino-button>
      <input class="ino-input-file__native-element"
             accept
             type="file"
             aria-hidden="true"
      >
    </div>
  </ino-input-file>
</div>
`;
